### PR TITLE
DOC: updating documentation to use Sphinx-7.x

### PR DESCRIPTION
**Pull request by Xarthisius**
## PR Summary

Infra related change: docs are now built in Ubuntu 23.04 container (py3.11)

### CODE REVIEW

**Review by neutrinoceros — DISMISSED** [review]

**Comment by neutrinoceros** [comment]
The image test failure is unrelated and already fixed on main

**Comment by Xarthisius** [comment]
> @Xarthisius assuming the docs build completes without errors, is this ready for review ?

Not yet. I'm still experimenting.

**Comment by Xarthisius** [comment]
@yt-fido test this please

**Comment by Xarthisius** [comment]
@yt-fido test this please.

**Comment by Xarthisius** [comment]
@yt-fido test this please.

**Comment by Xarthisius** [comment]
It's not fully fixed yet, but at least it passes and renders most notebooks. I think we can merge it and incrementally fix the remaining issues.

**Comment by neutrinoceros** [comment]
Can you open an issue to track what's left to be done ?

File: pyproject.toml [hunk 1]
```diff
@@ -192,14 +192,12 @@ doc = [
...
-    "nose==1.3.7",
+    "jupyter-client",
+    "nbsphinx",
+    "nose~=1.3.7; python_version < '3.10'",
     "pytest>=6.1",
     "pyx>=0.15",
-    "runnotebook",
-    "scipy<=1.7.1", # see https://github.com/yt-project/yt/issues/3966 and https://github.com/scipy/scipy/issues/16602
-    "sphinx==3.5.4",
+    "sphinx",
```
Comment (neutrinoceros) [inline pyproject.toml:200]: Can we set an explicit lower boundary to document what currently works ?

**Review by neutrinoceros — COMMENTED** [review]
I didn't get why certain rst files were removed, can you explain ?

**Comment by Xarthisius** [comment]
> I didn't get why certain rst files were removed, can you explain ?

They were just stubs invoking `::notebook` to render and link to a html version of the notebook. `nbsphinx` treats ipynb as native documents so you can now link directly.

File: pyproject.toml (hunk 1 above)
Comment (Xarthisius) [reply pyproject.toml]: Added current versions as lower bounds to all the packages in `[docs]`

**Review by jzuhone — APPROVED** [review]

**Comment by Xarthisius** [comment]
> Can you open an issue to track what's left to be done ?

I think the existing issue about droping runnotebook has it covered. I left a comment there.

**Comment by neutrinoceros** [comment]
Ok thanks, then I'll unlink this PR

**Pull request merged**